Skeletonize

jsf9k opened the pull request:
## 🗣 Description ##

This pull request skeletonizes the repo against [cisagov/skeleton-docker](https://github.com/cisagov/skeleton-docker).

## 💭 Motivation and context ##

All of our repos that are still being used should be based on a skeleton repo.

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release.

mcdonnnj committed:
Update pre-commit hooks

Hooks updated with the `pre-commit autoupdate` command.

Note:
The `ansible-lint` hook is intentionally being held back on 4.3.7 because of
ongoing issues with the 5.x version and how we use Ansible (standalone Galaxy
roles).

felddy committed:
Add context around command-line instructions for non-technical folks.

felddy committed:
Update README.md

Co-authored-by: Hillary <hillary.jeffrey@trio.dhs.gov>

mcdonnnj committed:
Add the tag.sh script to this repository

Since the GitHub Actions configuration builds images on tag pushes, we should
use the standard tagging script we use for the same purpose in other
repositories. This will make it easier to tag, build, and push Docker images
of a specific version in a standardized manner.

felddy committed:
Update README.md

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Update configuration instructions for pyenv

With the release of pyenv v2.0.0 there is a breaking change around startup
logic that necessitates an update for our pyenv setup instructions. Also
add a statement about how to get configuration instructions from pyenv
itself.

mcdonnnj committed:
Merge pull request #71 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit Hooks

mcdonnnj committed:
Merge branch 'develop' into maintenance/remove_user_from_codeowners

mcdonnnj committed:
Merge pull request #72 from cisagov/maintenance/remove_user_from_codeowners

Remove User from CODEOWNERS

mcdonnnj committed:
Merge branch 'develop' into documentation/update_pyenv_instructions

mcdonnnj committed:
Merge pull request #75 from cisagov/documentation/update_pyenv_instructions

Update Instructions for Configuring pyenv

mcdonnnj committed:
Merge branch 'develop' into improvement/gitignore_me

mcdonnnj committed:
Merge pull request #70 from cisagov/improvement/gitignore_me

Improve gitignore file comments and organization.

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Add Docker section to the gitignore file

Mirror changes in cisagov/skeleton-generic by adding a Docker section to the
gitignore file and moving appropriate entries to that section.

mcdonnnj committed:
Merge pull request #59 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/skip_codeql_on_push_for_dependabot_branches

mcdonnnj committed:
Merge pull request #53 from cisagov/improvement/skip_codeql_on_push_for_dependabot_branches

Skip CodeQL Workflow for Dependabot Branches on push Events

mcdonnnj committed:
Merge branch 'develop' into bug/add_semver_to_requirements

mcdonnnj committed:
Merge pull request #55 from cisagov/bug/add_semver_to_requirements

Add semver to Development Requirements File

mcdonnnj committed:
Merge branch 'develop' into improvement/add_tag_script

mcdonnnj committed:
Merge pull request #57 from cisagov/improvement/add_tag_script

Add the tag.sh Script

mcdonnnj committed:
Merge branch 'develop' into improvement/gh_registry

mcdonnnj committed:
Merge pull request #49 from cisagov/improvement/gh_registry

Enable container publication to GitHub Container Registry.

mcdonnnj committed:
Merge branch 'develop' into improvement/readme_build_instructions

mcdonnnj committed:
Merge pull request #51 from cisagov/improvement/readme_build_instructions

Add better build instructions to the README.

mcdonnnj committed:
Standardize the layout of the Lineage configuration file

jsf9k committed:
Add a section to the README for ports

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Update the README to reference the latest version of the Docker container

This change was discussed among @cisagov/team-ois here:
https://github.com/cisagov/gatherer/pull/60#pullrequestreview-683086244

Also correct git comments to use the imperative mood.

jsf9k committed:
Merge pull request #62 from cisagov/improvement/use-latest-version-number-in-readme

Update the README to reference the latest version of the Docker container

jsf9k committed:
Document the single exposed (but non-functional) port

Co-authored-by: Nick M <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Merge pull request #61 from cisagov/documentation/add-ports-section

Add a section to the README for ports

mcdonnnj committed:
Add style enforcement rules

Add rules to enforce ATX-closed headers, dashes for unordered list elements,
and `1.` for ordered list elements.

mcdonnnj committed:
Add rule for image headers

Add <h1> and <img> tags to the allowed list for MD033 (HTML elements) to
support using an image as the first thing in a markdown file (header image).

mcdonnnj committed:
Switch to a YAML markdownlint configuration file

This converts the existing `.mdl_config.json` file to an equivalent
`.mdl_config.yaml` file. The reference in the markdownlint pre-commit hook
configuration is updated to match.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Add comments to markdownlint configuration

Now that this is a YAML file we can add comments explaining the rule
modifications we use. This will make it easier to edit or expand in the future.

mcdonnnj committed:
Bump actions/github-script pin from v3 to v4

mcdonnnj committed:
Add the validate_manifest hook from pre-commit

This hook will validate any pre-commit hook manifest files in the repository.

dependabot[bot] committed:
Bump python from 3.9-alpine to 3.9.6-alpine

Bumps python from 3.9-alpine to 3.9.6-alpine.

---
updated-dependencies:
- dependency-name: python
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

jsf9k committed:
Install terraform and packer for the linting job

We should be doing this because the Packer and Terraform pre-commit
hooks leverage the corresponding executables; therefore, it makes
sense to go ahead and install the particular versions of those
executables that we support.  Also add support for optionally
debugging via tmate.

See also cisagov/skeleton-generic#74.

jsf9k committed:
Break out the curl cache creation into its own step

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Remove unnecessary line in tasks

There is no reason to create /usr/bin/terraform.  This is a vestige of
an earlier age.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Modify the Packer installation to model that of Terraform

The Terraform installation does not destroy the existing system
Terraform installation, and neither should the Packer installation.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Install a specific version of terraform-docs

Note that this change is dependent on the merging of
cisagov/setup-env-github-action#31.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Move go installation so that it takes place before the cache task

Some variables defined in the go installation are used in the cache
task, so the go installation must happen first.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Capitalize Go for consistency

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj committed:
Update pre-commit hooks

This is performed by running `pre-commit autoupdate`, but with the
`ansible-lint` hook held back manually.

jsf9k committed:
Prefer the newer "go install" syntax

As of [Go 1.16](https://tip.golang.org/doc/go1.16#go-command) the `GO111MODULE` environment variable defaults to `on` and `go get` has been deprecated for module installation.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj committed:
Install the shfmt tool for GHA

The `shfmt` tool does not ship on the GitHub Actions runners so we must install
it manually.

mcdonnnj committed:
Replace the beautysh hook with pre-commit-shfmt

We have had a difficult time with how beautysh parses some shellscripts. I went
in pursuit of an alternative and I believe shfmt to be a good alternative.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Apply changes from the shfmt pre-commit hook

dav3r committed:
Use preferred capitalization for Docker Hub

dav3r committed:
Satisfy linter by shortening a long line

dav3r committed:
Use preferred capitalization for Docker Hub

dav3r committed:
Capitalize Docker in README

dav3r committed:
Fix additional Docker and Docker Hub capitalizations

mcdonnnj committed:
Merge pull request #83 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit Hooks

mcdonnnj committed:
Merge branch 'develop' into improvement/add_pre-commit_hook

mcdonnnj committed:
Merge pull request #80 from cisagov/improvement/add_pre-commit_hook

Add the validate_manifest Hook to pre-commit Configuration

mcdonnnj committed:
Merge branch 'develop' into improvement/update_mdl_configuration

jsf9k committed:
Merge branch 'develop' into improvement/install-tf-and-packer-for-linting

mcdonnnj committed:
Merge pull request #79 from cisagov/improvement/update_mdl_configuration

Update markdownlint Configuration

mcdonnnj committed:
Merge branch 'develop' into improvement/install-tf-and-packer-for-linting

mcdonnnj committed:
Merge pull request #82 from cisagov/improvement/install-tf-and-packer-for-linting

Install terraform and packer for the linting job

mcdonnnj committed:
Merge branch 'develop' into improvement/replace_beautysh_with_shfmt

mcdonnnj committed:
Merge pull request #84 from cisagov/improvement/replace_beautysh_with_shfmt

Replace beautysh hook with cisagov/pre-commit-shfmt

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Satisfy pre-commit hooks

Update to reflect the new pre-commit configuration.

mcdonnnj committed:
Merge pull request #66 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/standardize_lineage_configuration

mcdonnnj committed:
Merge pull request #60 from cisagov/improvement/standardize_lineage_configuration

Standardize the Layout of the Lineage Configuration File

mcdonnnj committed:
Merge branch 'develop' into update/bump_github-script_to_v4

mcdonnnj committed:
Merge pull request #63 from cisagov/update/bump_github-script_to_v4

Update the Pin for actions/github-script to v4

mcdonnnj committed:
Merge branch 'develop' into dav3r-patch-1

jsf9k committed:
Merge branch 'develop' into dependabot/docker/python-3.9.6-alpine

mcdonnnj committed:
Merge pull request #65 from cisagov/dav3r-patch-1

Use preferred capitalization for Docker and Docker Hub

jsf9k committed:
Merge branch 'develop' into dependabot/docker/python-3.9.6-alpine

dav3r committed:
Fix a typo

jsf9k committed:
Remove support for the linux/s390x platform

Issue cisagov/skeleton-docker#69 has been created to remind us to
re-add it once it is again supported by the official Python Docker
image.

mcdonnnj committed:
Use the hashicorp/setup-terraform Action

Instead of manually installing a Terraform binary we can leverage the Action
provided by Hashicorp to do the same thing.

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with switching to v5.

mcdonnnj committed:
Add a markdownlint rule for horizontal rules

This sets a specific rule for MD035 (Horizontal rule style) instead of the
default value of "consistent".

mcdonnnj committed:
Add a markdownlint rule for code blocks

This sets a specific rule for MD046 (code block style) instead of the default
value of "consistent".

mcdonnnj committed:
Merge branch 'develop' into dav3r-patch-1

mcdonnnj committed:
Merge pull request #85 from cisagov/dav3r-patch-1

Fix a typo

mcdonnnj committed:
Merge branch 'develop' into improvement/specify_mdl_rules

mcdonnnj committed:
Merge pull request #89 from cisagov/improvement/specify_mdl_rules

Add markdownlint Rules to Enforce Specific Formatting

mcdonnnj committed:
Merge branch 'develop' into improvement/use_setup-terraform_in_gha_workflow

mcdonnnj committed:
Merge pull request #88 from cisagov/improvement/use_setup-terraform_in_gha_workflow

Use the hashicorp/setup-terraform Action to Install Terraform in GHA Workflow

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

jsf9k committed:
Revert "Remove support for the linux/s390x platform"

This reverts commit 3ebf37da6ea14dd4bbdb607271d8faa9e1484dd2.

There was finally a successful build of the official Python Docker
image upstream, so we can add this back in.

Resolves cisagov/skeleton-docker#69.

dav3r committed:
Enable terraform validate pre-commit hook

This hook should now work as expected in most, if not all of our repos 
now that we are finally updating to terraform 0.13.x (on our way to 
1.0.x).

dav3r committed:
Temporarily use cisagov/setup-env-github-action@improvement/support_tf_0.13

This change will be reverted when testing is completed.

mcdonnnj committed:
Merge pull request #70 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into dependabot/docker/python-3.9.6-alpine

mcdonnnj committed:
Merge pull request #64 from cisagov/dependabot/docker/python-3.9.6-alpine

Bump python from 3.9-alpine to 3.9.6-alpine

dav3r committed:
Remove lint job step to initialize Terraform directories

Initialization will now be done during the "terraform validate" step.

dav3r committed:
Revert "Temporarily use cisagov/setup-env-github-action@improvement/support_tf_0.13"

This reverts commit 6a7fbf07bd371d0493c523ce24647e5c04c77c03.

dav3r committed:
Merge pull request #90 from cisagov/improvement/enable_terraform_validate

Enable terraform validate pre-commit hook

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

dependabot[bot] committed:
Bump actions/github-script from 4 to 5

Bumps [actions/github-script](https://github.com/actions/github-script) from 4 to 5.
- [Release notes](https://github.com/actions/github-script/releases)
- [Commits](https://github.com/actions/github-script/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/github-script
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with switching to v5.

mcdonnnj committed:
Use an id when using cisagov/setup-env-github-action

mcdonnnj committed:
Use setup-env outputs for Terraform version

Change from using an environment variable to the outputs of the
cisagov/setup-env-github-action instead.

mcdonnnj committed:
Use setup-env outputs for Packer version

We use a `PACKER_VERSION` environment variable for the "Install Packer" step
that is populated from the `setup-env` outputs to get around `yamllint` lint
length limits.

mcdonnnj committed:
Use setup-env outputs for shfmt version

We change the "Install shfmt" step to use two local environment variables to
provide the package's URL and version to install. This allows us to work around
`yamllint` line length limits.

mcdonnnj committed:
Use setup-env outputs for Terraform-docs version

We change the "Install Terraform-docs" step to use two local environment
variables to provide the package's URL and version to install. This allows us
to work around `yamllint` line length limits.

mcdonnnj committed:
Store Go version as a step output

This changes from using an environment variable to using a step output to store
the Go version that is installed. This mirrors changes made to the other
program versions and how they're stored.

mcdonnnj committed:
Update the ansible-lint version for pre-commit

mcdonnnj committed:
Add an ansible-lint configuration file

This file is copied from cisagov/skeleton-ansible-role#85 and tweaked to pass
pre-commit. See here for for a list of the elements that can exist in this
file: https://ansible-lint.readthedocs.io/en/latest/configuring.html

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Add dependabot configuration for Terraform

This will configure `dependabot` to scan Terraform configurations if they exist
in a repository. If a repository stores a Terraform configuration in a sub-
directory this configuration will need to be modified or an additional
configuration must be added if there still exists a Terraform configuration in
the root directory.

dependabot[bot] committed:
Bump python from 3.9.6-alpine to 3.10.0-alpine

Bumps python from 3.9.6-alpine to 3.10.0-alpine.

---
updated-dependencies:
- dependency-name: python
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Update for actions/github-script API changes

Update code to reflect breaking changes in V5 of cisagov/github-script as
mentioned here:
https://github.com/actions/github-script#breaking-changes-in-v5

jsf9k committed:
Merge pull request #74 from cisagov/dependabot/docker/python-3.10.0-alpine

Bump python from 3.9.6-alpine to 3.10.0-alpine

mcdonnnj committed:
Merge pull request #91 from cisagov/improvement/pre-commit_update

Update `pre-commit` Hooks

mcdonnnj committed:
Merge pull request #93 from cisagov/improvement/update_ansible-lint_pre-commit_version

Update the `ansible-lint` Version in the pre-commit Configuration

mcdonnnj committed:
Merge pull request #94 from cisagov/improvement/add_dependabot_configuration_for_terraform

Add `dependabot` Configuration for Terraform

mcdonnnj committed:
Merge pull request #92 from cisagov/improvement/use_action_outputs

Use Outputs of cisagov/setup-env-github-action to Control Installed Program Versions

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Update ansible-lint configuration

Remove the manual exclusion of the `converge.yml` file in the default molecule
scenario. With the changes in cisagov/skeleton-ansible-role#88 this exclusion
is no longer necessary.

mcdonnnj committed:
Update ansible-lint configuration

Instead of excluding molecule configurations for systemd individually we
instead direct ansible-lint to process any of these configurations in
scenario directories as plain yaml files. This mirrors how ansible-lint
handles `molecule.yml` files in scenario directories by default.

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`.

dependabot[bot] committed:
Bump python from 3.10.0-alpine to 3.10.1-alpine

Bumps python from 3.10.0-alpine to 3.10.1-alpine.

---
updated-dependencies:
- dependency-name: python
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge pull request #97 from cisagov/maintenance/update_pre-commit_hooks

Update `pre-commit` hooks

mcdonnnj committed:
Merge branch 'develop' into improvement/update_ansible-lint_configuration

mcdonnnj committed:
Merge pull request #95 from cisagov/improvement/update_ansible-lint_configuration

Update `ansible-lint` Configuration

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge branch 'develop' into lineage/skeleton

mcdonnnj committed:
Merge pull request #72 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into dependabot/docker/python-3.10.1-alpine

mcdonnnj committed:
Merge pull request #75 from cisagov/dependabot/docker/python-3.10.1-alpine

Bump python from 3.10.0-alpine to 3.10.1-alpine

mcdonnnj committed:
Merge branch 'develop' into dependabot/github_actions/actions/github-script-5

mcdonnnj committed:
Merge pull request #73 from cisagov/dependabot/github_actions/actions/github-script-5

Bump actions/github-script from 4 to 5

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`.

mcdonnnj committed:
Add comment to Dependabot configuration

This comment explains that the configuration may have commented out
ignore directives that should be uncommented in downstream projects.

mcdonnnj committed:
Disable comments-indentation rule for yamllint

yamllint does not like it when you comment out pieces of dictionaries
in lists. Upcoming additions to the Dependabot configuration will run
afoul of this so we are updating the yamllint configuration.

mcdonnnj committed:
Merge pull request #99 from cisagov/improvement/update_dependabot_and_yamllint_configurations

Add boilerplate for upcoming Dependabot configuration changes

mcdonnnj committed:
Merge branch 'develop' into maintenance/update_pre-commit_hooks

mcdonnnj committed:
Merge pull request #98 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit hooks

mcdonnnj committed:
Use asterisks for emphasis- and strong-styles

Given the inconsistent behavior of underscores used within words for
style we should prefer asterisks now that these rules are available.

mcdonnnj committed:
Add missing hyphen in markdownlint rule header

mcdonnnj committed:
Merge pull request #100 from cisagov/improvement/update_markdownlint_configuration

Use new `markdownlint` rules for emphasis- and strong-styles

dependabot[bot] committed:
Bump actions/setup-python from 2 to 3

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 2 to 3.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/checkout from 2 to 3

Bumps [actions/checkout](https://github.com/actions/checkout) from 2 to 3.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with upgrading to v6.

mcdonnnj committed:
Add Dependabot ignore directives

This adds commented out ignore directives for the following GitHub
Actions:

- action/cache
- action/checkout
- action/setup-python

These should be uncommented downstream to ensure that updates to these
dependencies are pushed from pull requests made in the skeleton.

mcdonnnj committed:
Upgrade from Python 3.9 to 3.10 for the `lint` job in GHA

mcdonnnj committed:
Use consistent quoting for software versions

mcdonnnj committed:
Merge pull request #101 from cisagov/dependabot/github_actions/actions/setup-python-3

Bump actions/setup-python from 2 to 3

mcdonnnj committed:
Merge branch 'develop' into dependabot/github_actions/actions/checkout-3

mcdonnnj committed:
Merge pull request #102 from cisagov/dependabot/github_actions/actions/checkout-3

Bump actions/checkout from 2 to 3

dependabot[bot] committed:
Bump actions/cache from 2 to 3

Bumps [actions/cache](https://github.com/actions/cache) from 2 to 3.
- [Release notes](https://github.com/actions/cache/releases)
- [Commits](https://github.com/actions/cache/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge branch 'develop' into improvement/add_dependabot_ignores_for_github_actions

mcdonnnj committed:
Merge pull request #107 from cisagov/improvement/add_dependabot_ignores_for_github_actions

Add Dependabot ignore directives

mcdonnnj committed:
Merge pull request #104 from cisagov/dependabot/github_actions/actions/cache-3

Bump actions/cache from 2 to 3

mcdonnnj committed:
Merge branch 'develop' into improvement/use_python_3.10_for_gha

mcdonnnj committed:
Merge pull request #106 from cisagov/improvement/use_python_3.10_for_gha

Use Python 3.10 for the `lint` job in GitHub Actions

mcdonnnj committed:
Merge branch 'develop' into maintenance/update_pre-commit_hooks

mcdonnnj committed:
Merge pull request #105 from cisagov/maintenance/update_pre-commit_hooks

Update `pre-commit` hooks

mcdonnnj committed:
Add pre-commit hooks to validate GitHub Actions content

Add hooks to validate GitHub Actions and GitHub Action workflows. These
are validated using JSON Schemas.

mcdonnnj committed:
Merge pull request #109 from cisagov/improvement/add_github_actions_pre-commit_hooks

Add `pre-commit` hooks to validate GitHub Actions content

mcdonnnj committed:
Add an ownership comment to the Dependabot configuration

Add a comment that states that the commented out ignore directives are
managed by cisagov/skeleton-generic.

mcdonnnj committed:
Add additional Dependabot ignore directives

This adds the other versioned Actions that should be managed by
cisagov/skeleton-generic to the list of commented out dependencies to
ignore.

mcdonnnj committed:
Update actions/setup-go from v2 to v3

Additionally as of v3.1.0 of actions/setup-go there is a go-version
output value to retrieve the version of Go installed by the Action.
This allows us to remove the step to manually retrieve this information
from the Go executable.

mcdonnnj committed:
Update the version of Go used from 1.16 to 1.18

Go 1.16 is no longer supported as of the release of 1.18 so it makes
sense to update to the latest version available.

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with upgrading to v6.

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Enable Dependabot ignore directives

Uncomment the `github-actions` Dependabot ignore directives because
those dependencies are managed by cisagov/skeleton-generic upstream.

mcdonnnj committed:
Fix issues with the `build.yml` file

We must ensure that all `workflow_dispatch` input defaults are string
values. This ensures expected behavior instead of relying on automatic
type conversions.

mcdonnnj committed:
Bump docker/build-push-action from 2 to 3

mcdonnnj committed:
Bump docker/login-action from 1 to 2

mcdonnnj committed:
Bump docker/setup-buildx-action from 1 to 2

mcdonnnj committed:
Bump docker/setup-qemu-action from 1 to 2

mcdonnnj committed:
Bump actions/download-artifact from 2 to 3

mcdonnnj committed:
Bump actions/upload-artifact from 2 to 3

mcdonnnj committed:
Add Dependabot dependency ownership for this repository

This includes adding the versioned GitHub Actions that should be
ignored in downstream projects and a comment stating where these
dependencies are managed.

jsf9k committed:
Update code to use the "docker compose" syntax vice "docker-compose"

The "docker compose" syntax is the preferred (and only correct) syntax
after the changes in cisagov/ansible-role-docker#60.

mcdonnnj committed:
Merge pull request #79 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge pull request #101 from cisagov/improvement/update_dependabot_ignores

Update Dependabot ignore directives

mcdonnnj committed:
Merge pull request #102 from cisagov/improvement/update_docker_action_versions

Update GitHub Action versions for this project

jsf9k committed:
Improve language in a comment

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj committed:
Merge branch 'develop' into bugfix/update-docker-compose-syntax

mcdonnnj committed:
Merge pull request #105 from cisagov/bugfix/update-docker-compose-syntax

Update code to use the "docker compose" syntax vice "docker-compose"

mcdonnnj committed:
Remove dangling argument in the pre-commit configuration

This argument references a file that is no longer found in the
repository. The configuration also no longer uses any such files so it
is safe to remove this as opposed to updating it instead.

mcdonnnj committed:
Update files used for Python requirements by LGTM

Add all requirements*.txt files in the configuration to this file. This
ensures that all requirements in use are being used for scanning.

mcdonnnj committed:
Restore the alerts badging for LGTM

Until we completely remove LGTM from our repositories we should keep at
least the minimum badging. This project does not directly use any
Python, JavaScript, etc. so I am enabling the "Total alerts" badge.
Other possibilities are commented out to make it easier for
downstream repositories to enable what suits their configurations.

mcdonnnj committed:
Enable the allow-non-breakable-inline-mappings rule for yamllint

mcdonnnj committed:
Increase the maximum line length in the yamllint configuration

Increase the line length maximum from the default of 80 to 88. This 10%
increase mirrors the line length allowed in other linter configurations
such as black.

jsf9k committed:
Fix overly match-happy sed commands

We saw in cisagov/postfix-docker#47 that the sed commands in the
bump_version.sh script could inadvertently match the CC0 version in
the README.md file.  This change escapes the periods in the version
before passing it on to sed so that they only match periods and not
just any character.

mcdonnnj committed:
Add a label configuration file

This configuration file stores information about the labels expected in
this repository.

mcdonnnj committed:
Add a workflow to synchronize repository labels

This adds a workflow to ensure that the repository labels are updated
to reflect changes to the label configuration file .github/labels.yml.

mcdonnnj committed:
Remove usage of set-output from our GitHub Actions workflow

GitHub has deprecated the set-output command per:
https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

This updates the GitHub Actions workflow to use the newly preferred
method to set the output for a job's step.

mcdonnnj committed:
Fix the repo of the flake8 pre-commit hook

Per PyCQA/flake8#1290 this hook moved from GitLab to GitHub. The
version we use is bumped to the latest tag on GitHub as well.

jsf9k committed:
Remove duplicate label

jsf9k committed:
Add a label for issues or PRs that involve test code

jsf9k committed:
Prettify a comment

jsf9k committed:
Rework creation of unprivileged user and installation of dependencies

Also add a section to update pip and setuptools via pip.

jsf9k committed:
Add comments and prefer long CLI options

jsf9k committed:
Move ECHO_MESSAGE declaration to the "Prepare to run" section

jsf9k committed:
Merge remote-tracking branch 'skeleton-docker/develop' into skeletonize

dav3r commented on README.md:
```diff
@@ -1,6 +1,14 @@
-# NCATS pshtt, trustymail, and sslyze Saver :floppy_disk: #
+# NCATS pshtt, trustymail, and sslyze Saver #
```
NCATS is no more... 😢 
```suggestion
# pshtt, trustymail, and sslyze Saver #
```

dav3r commented on README.md:
```diff
@@ -12,9 +20,236 @@ database](https://www.mongodb.com/)
+| Image:tag | Description |
+|-----------|-------------|
+|`cisagov/saver:1.3.0`| An exact release version. |
+|`cisagov/saver:1.2`| The most recent release matching the major and minor version numbers. |
```
Correct version number:
```suggestion
|`cisagov/saver:1.3`| The most recent release matching the major and minor version numbers. |
```

dav3r commented on src/pshtt_csv2mongo.py:
```diff
@@ -0,0 +1,224 @@
+        writer.writerow([agency])
+
+    # Create the clean-current-federal for use later.
+    # TODO: Remove this and have it handled in code.
```
I'm not really clear on what this TODO means, but if it's still valid, you should create an issue for it and reference it here.

jsf9k replied:
Please see commit a17deb4.

dav3r commented on src/sslyze_csv2mongo.py:
```diff
@@ -0,0 +1,237 @@
+        writer.writerow([agency])
+
+    # Create the clean-current-federal for use later.
+    # TODO: Remove this and have it handled in code.
```
If this is still valid, you should create an issue for it and reference it here.

jsf9k replied:
Please see commit a17deb4.

dav3r commented on src/trustymail_csv2mongo.py:
```diff
@@ -0,0 +1,250 @@
+        writer.writerow([agency])
+
+    # Create the clean-current-federal for use later.
+    # TODO: Remove this and have it handled in code.
```
If this is still valid, you should create an issue for it and reference it here.

jsf9k replied:
Please see commit a17deb4.

dav3r commented on src/trustymail_csv2mongo.py:
```diff
@@ -0,0 +1,250 @@
+        'Successfully imported {} documents to "{}" database on '
+        "{}".format(domains_processed, db.name, db.client.address[0])
+    )
+    # import IPython; IPython.embed() #<<< BREAKPOINT >>>
+
```
This can be deleted.
```suggestion
```

dav3r reviewed (changes requested):
Just a few minor points to be resolved here.

jsf9k committed:
Remove mention of NCATS

Co-authored-by: dav3r <daver@geekpad.com>

jsf9k committed:
Correct version number in README

Co-authored-by: dav3r <daver@geekpad.com>

jsf9k committed:
Remove useless line

Co-authored-by: dav3r <daver@geekpad.com>

jsf9k committed:
Remove repeated TODO lines

These TODOs were bequeathed to me when I inherited this code, but I
have no idea what they are referring to.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

dav3r reviewed (approved):
Strong work! 💪 💼

mcdonnnj committed:
Update the version of Go used from 1.18 to 1.19

Go 1.19 was released while this branch was in the wings and it makes
sense to bump to the latest Go release.

mcdonnnj committed:
Merge pull request #113 from cisagov/improvement/update_go_configuration

Update Go installation in the `build.yml` workflow

dependabot[bot] committed:
Bump hashicorp/setup-terraform from 1 to 2

Bumps [hashicorp/setup-terraform](https://github.com/hashicorp/setup-terraform) from 1 to 2.
- [Release notes](https://github.com/hashicorp/setup-terraform/releases)
- [Changelog](https://github.com/hashicorp/setup-terraform/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/setup-terraform/compare/v1...v2)

---
updated-dependencies:
- dependency-name: hashicorp/setup-terraform
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge pull request #111 from cisagov/dependabot/github_actions/hashicorp/setup-terraform-2

Bump hashicorp/setup-terraform from 1 to 2

dependabot[bot] committed:
Bump actions/setup-python from 3 to 4

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 3 to 4.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge pull request #115 from cisagov/dependabot/github_actions/actions/setup-python-4

Bump actions/setup-python from 3 to 4

mcdonnnj committed:
Merge pull request #112 from cisagov/improvement/update_dependabot_ignores

Update Dependabot ignore directives

mcdonnnj committed:
Merge pull request #119 from cisagov/bug/update_github_actions_workflow

Remove usage of `set-output` from our GitHub Actions workflow

mcdonnnj committed:
Merge pull request #117 from cisagov/improvement/update_yamllint_line-length_rules

Update the `line-length` configuration for `yamllint`

mcdonnnj committed:
Merge pull request #120 from cisagov/bug/verify_pre-commit_hook_repos

Update the `flake8` pre-commit hook configuration

mcdonnnj committed:
Merge pull request #114 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit hooks

mcdonnnj committed:
Merge pull request #118 from cisagov/improvement/add_label_sync_workflow

Add workflow to manage repository labels

mcdonnnj committed:
Fix GitHub label color values

Update the configuration for repository labels to remove the leading
`#` from color values. With a `#` leading the values they are seen as
invalid by the GitHub API.

mcdonnnj committed:
Correct spacing in yamllint rules

There was a missing empty line in the `.yamllint` file between two rule
definitions.

mcdonnnj committed:
Merge pull request #122 from cisagov/bug/fix_bad_configurations

Update two configuration files

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into lineage/skeleton

mcdonnnj committed:
Add repository specific label

mcdonnnj committed:
Update Python setup in the test job for GHA

Bump actions/setup-python from v3 to v4 and use Python 3.10 to mirror
the `lint` job.

mcdonnnj committed:
Update the CodeQL configuration

Pull in the CodeQL workflow from cisagov/skeleton-python-library to
update the CodeQL configuration for this repository. We also add
github/codeql-action to the dependencies managed by this repository in
the dependabot configuration.

mcdonnnj committed:
Remove uses of the `set-output` command

Remove the remaining uses of the `set-output` command since it has been
deprecated per:
https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

mcdonnnj committed:
Update the version of the second bandit hook

The `bandit (everything else)` hook was not updated in sync with the
`bandit (tests tree)` hook and is using an older version of bandit.

mcdonnnj committed:
Correct poor English in comments for the bandit pre-commit hooks

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj committed:
Merge pull request #136 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Merge pull request #138 from cisagov/bug/update_bandit_in_pre-commit_configuration

Update the version of the second `bandit` hook

mcdonnnj committed:
Merge pull request #137 from cisagov/improvement/update_codeql_workflow

Update the CodeQL configuration

mcdonnnj committed:
Merge pull request #115 from cisagov/improvement/fix_dangling_configurations

Update some outdated items in the project

dependabot[bot] committed:
Bump actions/github-script from 5 to 6

Bumps [actions/github-script](https://github.com/actions/github-script) from 5 to 6.
- [Release notes](https://github.com/actions/github-script/releases)
- [Commits](https://github.com/actions/github-script/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/github-script
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge pull request #77 from cisagov/dependabot/github_actions/actions/github-script-6

Bump actions/github-script from 5 to 6

mcdonnnj committed:
Merge pull request #122 from cisagov/bugfix/tighten-regex-in-version-script

Fix overly eager sed commands

mcdonnnj committed:
Merge pull request #135 from cisagov/improvement/dockerfile

Improve Dockerfile

mcdonnnj committed:
Remove LGTM from the repository

LGTM.com is shutting down 2022-12-16 so we should remove all
LGTM-related items to prepare for disabling the LGTM.com integration in
the organization.

mcdonnnj committed:
Merge pull request #139 from cisagov/maintenance/remove_lgtm

Remove LGTM from the repository

mcdonnnj committed:
Restore Dockerfile changes from review

I accidentally clobbered these changes when I was updating
cisagov/skeleton-docker#135 for merge. This restores the changes made
during review as well as fixing a missed reference to the CISA_USER
argument.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Merge pull request #140 from cisagov/bug/fix_clobbered_dockerfile

Restore Dockerfile changes from review

jsf9k committed:
Merge remote-tracking branch 'skeleton-docker/develop' into skeletonize

dav3r reviewed (approved):
Approval continues... 🥇

jsf9k committed:
Make minor changes to satisfy pyupgrade linter

jsf9k committed:
Add TODO comments for cisagov/orchestrator#130

jsf9k committed:
Remove unnecessary ARG lines

jsf9k committed:
Bump version from 1.3.0 to 1.3.0-rc.1

mcdonnnj commented on .github/labels.yml:
```diff
@@ -0,0 +1,70 @@
+- color: "fcdb45"
+  description: This pull request is awaiting an action or decision to move forward
+  name: on hold
+- color: "ef476c"
```
Since there are a number of Python pieces in this Docker image.

```suggestion
- color: "3772a4"
  description: Pull requests that update Python code
  name: python
- color: "ef476c"
```

dav3r reviewed (approved):
LGTM - nice work! 👍

mcdonnnj commented on Dockerfile:
```diff
@@ -38,58 +63,25 @@ RUN apk --no-cache --quiet add $DEPS
-# Setup working directory and entrypoint
 ###
-FROM setup_user AS final
+COPY src ${CISA_HOME}
```
So we only bring in the files we actually need.

```suggestion
COPY src/*.py src/*.sh ${CISA_HOME}
COPY src/include ${CISA_HOME}/include
```

mcdonnnj commented on Dockerfile:
```diff
@@ -38,58 +63,25 @@ RUN apk --no-cache --quiet add $DEPS
-RUN apk --no-cache --quiet del $SETUP_USER_DEPS
-
 # Put this just before we change users because the copy (and every
 # step after it) will always be rerun by docker, but we need to be
```
```suggestion
# step after it) will always be rerun by Docker, but we need to be
```

mcdonnnj commented on README.md:
```diff
@@ -12,9 +20,236 @@ database](https://www.mongodb.com/)
+
+| Filename      | Purpose              |
+|---------------|----------------------|
+| database_creds.yml | Cyber Hygiene read-only database credentials in [this format](https://github.com/cisagov/mongo-db-from-config#usage) |
```
```suggestion
| database_creds.yml | Cyber Hygiene read-only database credentials in [this format](https://github.com/cisagov/mongo-db-from-config#usage). |
```

mcdonnnj commented on Dockerfile:
```diff
@@ -1,13 +1,39 @@
+    && adduser --system --uid ${CISA_UID} --ingroup ${CISA_GROUP} ${CISA_USER}
 
 ###
 # Dependencies
```
Will you verify that `bash` is needed by this image? The `save_to_db.sh` script does not appear to need it per https://github.com/cisagov/saver/blob/b1b556cfa703e28a9ee09f87bc35c7813d2591c8/src/save_to_db.sh#L1 If it is required will you update this comment block to include a blurb about `bash` like we did in https://github.com/cisagov/scanner/commit/d1d67c011d4824f6d308fd71d92831040c6c174c?

jsf9k replied:
You are correct.  `bash` is not required.  Please see commit f8a068a.

jsf9k committed:
Add a python label

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Add a period at the end of a sentence

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Copy into the image only what we need

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Capitalize Docker

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Remove bash as a dependency

Bash is not required for this Docker container, since the script
save_to_db.sh uses /bin/sh.  Unlike bash, sh _is_ provided by default
by Alpine Linux.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Finalize version from 1.3.0-rc.1 to 1.3.0

mcdonnnj commented on Dockerfile:
```diff
@@ -1,13 +1,39 @@
+FROM python:3.10.1-alpine
```
This matches the version of Python on `cisagov/saver:1.2.5`.

```suggestion
FROM python:3.10.8-alpine
```

jsf9k replied:
Good catch!

mcdonnnj reviewed (approved):
One item I missed in previous review but otherwise LGTM ✔

jsf9k committed:
Revert accidental decrement of  base image version

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>
